### PR TITLE
feat(create-cloudflare): update submenu userflow

### PR DESCRIPTION
**Pull request by edmundhung**
## What this PR solves / how to test

This PR refines the questions asked in c3:

First of all, we will ask for the kind of templates that should be created.

Screenshot 2024-07-26 at 15 08 35

Depending on the category selected, we will show a different set of templates.

Screenshot 2024-07-26 at 15 43 11

Screenshot 2024-07-26 at 15 08 59

If the template selected supports different languages, we will also ask for the language user preferred.

Screenshot 2024-07-26 at 15 09 16

Two new arguments will also be supported:

```
--category=<value>
    Specifies the kind of templates that should be created

    Allowed Values:

      hello-world
        Hello World example
      web-framework
        Framework Starter
      demo
        Demo application
      remote-template
        Template from a Github repo

--lang=<value>
    The programming language of the template

    Allowed Values:

      ts, js, python
```

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required / Maybe required
  - [ ] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [x] Cloudflare docs PR(s): https://github.com/cloudflare/cloudflare-docs/pull/15825
  - [ ] Not necessary because:

### CODE REVIEW

File: packages/create-cloudflare/templates/hello-world-python/c3.ts [hunk 1]
```diff
@@ -1,9 +0,0 @@
-export default {
-	configVersion: 1,
-	id: "hello-world-python",
-	displayName: '"Hello World" Worker (Python)',
-	platform: "workers",
-	copyFiles: {
-		path: "./py",
-	},
-};
```
Comment (RamIdeas) [inline packages/create-cloudflare/templates/hello-world-python/c3.ts:9]: Is this deletion intentional?

**Review by RamIdeas — APPROVED** [review]

File: packages/create-cloudflare/templates/hello-world-python/c3.ts (hunk 1 above)
Comment (edmundhung) [reply packages/create-cloudflare/templates/hello-world-python/c3.ts:9]: Yes. The hello-world-python template is merged into the hello-world template.

Comment (RamIdeas) [reply packages/create-cloudflare/templates/hello-world-python/c3.ts:9]: Maybe I'm missing something but only the config file has been deleted – should the whole template be deleted?

Comment (edmundhung) [reply packages/create-cloudflare/templates/hello-world-python/c3.ts:9]: They are moved inside the `hello-world` folder with the `python` variant [added to the config](https://github.com/cloudflare/workers-sdk/pull/6320/files#diff-68d52b4c5d74735efdc8dcfe9606b81934e4f0f7646e7acac09b3cd0c3455dccR14-R16).

**Pull request merged**